### PR TITLE
Add Configuration Reference

### PR DESCRIPTION
**Pull request by gregurco**
| Q                | A
| ---------------- | -----
| Bug fix          | no
| New feature      | no
| BC breaks        | no
| Deprecations     | no
| Tests pass       | yes
| Fixed tickets    | -
| License          | MIT

It's good practice to have one article page with full configuration reference and on the main page (readme.md) to keep just the most used configuration parameters.

See docs: https://github.com/gregurco/EightPointsGuzzleBundle/blob/configuration_reference/src/Resources/doc/configuration-reference.md

Example:
[FOSRestBundle](https://symfony.com/doc/current/bundles/FOSRestBundle/configuration-reference.html)
[FOSUserBundle](https://symfony.com/doc/current/bundles/FOSUserBundle/configuration_reference.html)
[DoctrineBundle](https://symfony.com/doc/current/bundles/DoctrineBundle/configuration.html)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,20 +52,16 @@ To use the newest (maybe unstable) version please add following into your compos
...
-_Note: for symfony 3.3 and 3.4 you should install symfony/flex by yourself. From 4.0 it is included be default._
+_Note: we created [Symfony Flex Recipe][15] to speed up the installation of package._
 
 ----
 
 ## Usage
+
 ##### Load bundle in AppKernel.php:
+
+*Skip it for Symfony >= 4.0*
```
Comment (florianpreusner) [inline README.md]: what do you think about moving legacy (<= 4.0) to a separate README.md? So we can focus on the shiny new things in the main README.md and still provide information for older versions.

Comment (gregurco) [reply README.md]: Good idea. What do you think about link to https://github.com/8p/EightPointsGuzzleBundle/blob/v7.3.1/README.md ? Or you thought about totally another file?

File: src/Resources/doc/configuration-reference.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
+# Configuration Reference
+
+``` yaml
+eight_points_guzzle:
+    # (de)activate logging/profiler; default: %kernel.debug%
+    logging: true
```
Comment (rpkamp) [inline src/Resources/doc/configuration-reference.md]: `profiling` option is missing here, plus when https://github.com/8p/EightPointsGuzzleBundle/pull/228 lands that should be added as well :smiley:

Comment (gregurco) [reply src/Resources/doc/configuration-reference.md]: right :slightly_smiling_face: I also plan to add `slow_response_time` after merging your PR.

File: src/Resources/doc/configuration-reference.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+    clients:
+        api_payment:
```
Comment (rpkamp) [inline src/Resources/doc/configuration-reference.md]: I've also seen some packages that have a `minimum workable config", maybe that would be nice to add here as well. For example, in a project where I use this I just have this:

```
eight_points_guzzle:
  clients:
    api_payment: ~
```

and that works just fine. Just to show you don't _have_ pass any parameters if you don't want to.

File: src/Resources/doc/configuration-reference.md (hunk 2 above)
Comment (gregurco) [reply src/Resources/doc/configuration-reference.md]: added `profiling` and `slow_response_time`

File: src/Resources/doc/configuration-reference.md (hunk 3 above)
Comment (gregurco) [reply src/Resources/doc/configuration-reference.md]: Good idea :+1: did it.

**Pull request merged**